### PR TITLE
Try to parse date string with ISO8601 if previous conversion failed

### PR DESCRIPTION
**Pull request by johndelong**
Try to parse date string with ISO8601 if previous conversion failed

### CODE REVIEW

File: RSSParser/RSSParser.m [hunk 1]
```diff
@@ -199,6 +211,27 @@ -(NSURL *)getNSURLFromString:(NSString *)string {
...
+        if (!date) {
+            date = [_ISO8601formatter dateFromString:string];
+        }
+        if (!date) {
+            NSRange range = [string rangeOfString:@" " options:NSBackwardsSearch];
+            NSString *threeLetterZone = [string substringFromIndex:range.location+1];
+            NSTimeZone *timeZone = [NSTimeZone timeZoneWithAbbreviation:threeLetterZone];
+            if (timeZone)
+            {
+                NSString *gmtTime = [string stringByReplacingOccurrencesOfString:threeLetterZone withString:@"GMT"];
```
Comment (jkloo) [inline RSSParser/RSSParser.m:229]: What is the purpose of this?

File: RSSParser/RSSParser.m [hunk 2]
```diff
@@ -23,11 +25,19 @@ @implementation RSSParser
...
-        items = [[NSMutableArray alloc] init];
+        items = [NSMutableArray new];
         
-        _formatter = [[NSDateFormatter alloc] init];
+        _formatter = [NSDateFormatter new];
         [_formatter setLocale:[[NSLocale alloc] initWithLocaleIdentifier:@"en_EN"]];
         [_formatter setDateFormat:@"EEE, dd MMM yyyy HH:mm:ss Z"];
+
+        _formatterWithTimeZoneAbbreviation = [NSDateFormatter new];
+        [_formatterWithTimeZoneAbbreviation setLocale:[[NSLocale alloc] initWithLocaleIdentifier:@"en_EN"]];
```
Comment (jkloo) [inline RSSParser/RSSParser.m]: consider only creating 1 NSLocale object, considering they are all the same

File: RSSParser/RSSParser.m [hunk 3]
```diff
@@ -199,6 +211,27 @@ -(NSURL *)getNSURLFromString:(NSString *)string {
     }
 }
 
+-(NSDate *)getDateFromString:(NSString *)string {
+    NSDate *date = nil;
+    if (string) {
+        date = [_formatter dateFromString:string];
```
Comment (jkloo) [inline RSSParser/RSSParser.m]: you should use property notation outside of the `init` (e.g. `self.formatter`)

File: RSSParser/RSSParser.m (hunk 1 above)
Comment (johndelong) [inline RSSParser/RSSParser.m:229]: I don't know why, but for the life of me, I could not get an NSDate object if the `dateFormatString` contained "EST".

Even so, the example I found online used this method to convert a time to UTC and it seemed to work so...(shrug)

Comment (johndelong) [inline RSSParser/RSSParser.m:229]: oh, also there is only `secondsFromGMT` on a `TimeZone` object. There is no such thing as `secondsFromEST`

File: RSSParser/RSSParser.m (hunk 2 above)
Comment (johndelong) [inline RSSParser/RSSParser.m]: Good idea

File: RSSParser/RSSParser.m (hunk 3 above)
Comment (johndelong) [inline RSSParser/RSSParser.m]: grrrr - i know better...

**Pull request merged**